File: animated-base/src/test/java/com/facebook/fresco/animation/bitmap/wrapper/AnimatedDrawableBackendAnimationInformationTest.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.fresco.animation.bitmap.wrapper;

import static org.fest.assertions.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import com.facebook.fresco.animation.backend.AnimationBackend;
import com.facebook.imagepipeline.animated.base.AnimatedDrawableBackend;
import org.junit.Before;
import org.junit.Test;

/**
 * Tests {@link AnimatedDrawableBackendAnimationInformation}.
 */
public class AnimatedDrawableBackendAnimationInformationTest {

  private AnimatedDrawableBackend mAnimatedDrawableBackend;
  private AnimatedDrawableBackendAnimationInformation mAnimatedDrawableBackendAnimationInformation;

  @Before
  public void setup() {
    mAnimatedDrawableBackend = mock(AnimatedDrawableBackend.class);
    mAnimatedDrawableBackendAnimationInformation =
        new AnimatedDrawableBackendAnimationInformation(mAnimatedDrawableBackend);
  }

  @Test
  public void testGetFrameCount() throws Exception {
    when(mAnimatedDrawableBackend.getFrameCount()).thenReturn(123);

    assertThat(mAnimatedDrawableBackendAnimationInformation.getFrameCount()).isEqualTo(123);
  }

  @Test
  public void testGetFrameDurationMs() throws Exception {
    when(mAnimatedDrawableBackend.getDurationMsForFrame(1)).thenReturn(123);
    when(mAnimatedDrawableBackend.getDurationMsForFrame(2)).thenReturn(200);

    assertThat(mAnimatedDrawableBackendAnimationInformation.getFrameDurationMs(1)).isEqualTo(123);
    assertThat(mAnimatedDrawableBackendAnimationInformation.getFrameDurationMs(2)).isEqualTo(200);
  }

  @Test
  public void testGetLoopCount() throws Exception {
    when(mAnimatedDrawableBackend.getLoopCount()).thenReturn(123);

    assertThat(mAnimatedDrawableBackendAnimationInformation.getLoopCount()).isEqualTo(123);
  }

  @Test
  public void testGetLoopCountInfinite() throws Exception {
    when(mAnimatedDrawableBackend.getLoopCount()).thenReturn(AnimationBackend.LOOP_COUNT_INFINITE);

    assertThat(mAnimatedDrawableBackendAnimationInformation.getLoopCount())
        .isEqualTo(AnimationBackend.LOOP_COUNT_INFINITE);
  }
}

File: animated-base/src/test/java/com/facebook/fresco/animation/bitmap/wrapper/AnimatedDrawableBackendAnimationInformationTest.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.animation.bitmap.wrapper

import com.facebook.fresco.animation.backend.AnimationBackend
import com.facebook.imagepipeline.animated.base.AnimatedDrawableBackend
import org.assertj.core.api.Assertions
import org.junit.Before
import org.junit.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.whenever

/** Tests [AnimatedDrawableBackendAnimationInformation]. */
class AnimatedDrawableBackendAnimationInformationTest {
  private lateinit var animatedDrawableBackend: AnimatedDrawableBackend
  private lateinit var animatedDrawableBackendAnimationInformation:
      AnimatedDrawableBackendAnimationInformation

  @Before
  fun setup() {
    animatedDrawableBackend = mock()
    animatedDrawableBackendAnimationInformation =
        AnimatedDrawableBackendAnimationInformation(animatedDrawableBackend)
  }

  @Test
  @Throws(Exception::class)
  fun testGetFrameCount() {
    whenever(animatedDrawableBackend.getFrameCount()).thenReturn(123)

    Assertions.assertThat(animatedDrawableBackendAnimationInformation.getFrameCount())
        .isEqualTo(123)
  }

  @Test
  @Throws(Exception::class)
  fun testGetFrameDurationMs() {
    whenever(animatedDrawableBackend.getDurationMsForFrame(1)).thenReturn(123)
    whenever(animatedDrawableBackend.getDurationMsForFrame(2)).thenReturn(200)

    Assertions.assertThat(animatedDrawableBackendAnimationInformation.getFrameDurationMs(1))
        .isEqualTo(123)
    Assertions.assertThat(animatedDrawableBackendAnimationInformation.getFrameDurationMs(2))
        .isEqualTo(200)
  }

  @Test
  @Throws(Exception::class)
  fun testGetLoopCount() {
    whenever(animatedDrawableBackend.getLoopCount()).thenReturn(123)
    Assertions.assertThat(animatedDrawableBackendAnimationInformation.getLoopCount()).isEqualTo(123)
  }

  @Test
  @Throws(Exception::class)
  fun testGetLoopCountInfinite() {
    whenever(animatedDrawableBackend.getLoopCount())
        .thenReturn(AnimationBackend.LOOP_COUNT_INFINITE)

    Assertions.assertThat(animatedDrawableBackendAnimationInformation.getLoopCount())
        .isEqualTo(AnimationBackend.LOOP_COUNT_INFINITE)
  }
}
